basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -7,8 +7,107 @@
 import static org.junit.Assert.*;
 
 public class LibraryTest {
-    @Test public void testSomeLibraryMethod() {
+
+    //dice tests
+    @Test
+    public void testRoll() {
+        Library classUnderTest = new Library();
+        int input = 4;
+        int expectedOutput = 4;
+        assertEquals("rollMethod return array of dice rolls of size n", expectedOutput, classUnderTest.roll(input).length );
+    }
+    @Test
+    public void testRoll_noRolls() {
+        Library classUnderTest = new Library();
+        int input = 0;
+        int expectedOutput = 0;
+        assertEquals("rollMethod return array of dice rolls of size n", expectedOutput, classUnderTest.roll(input).length );
+    }
+
+    //duplicates tests
+    @Test
+    public void testContainsDuplicate_duplicates() {
+        Library classUnderTest = new Library();
+        int[] input = new int[]{1, 2, 3, 2, 5};
+        assertTrue(" containsDuplicateMethod should return 'true' if the array has duplicate values and false if not", classUnderTest.containsDuplicate(input));
+    }
+
+    @Test
+    public void testContainsDuplicate_noDuplicates() {
+        Library classUnderTest = new Library();
+        int[] input = new int[]{1, 2, 3, 4, 5};
+        assertFalse(" containsDuplicateMethod should return 'true' if the array has duplicate values and false if not", classUnderTest.containsDuplicate(input));
```

Comment:
Copy & Paste alert! The description says this should return 'true' when it should say it returns false